clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -87,7 +87,7 @@ export class HttpConnection implements IConnection {
                 // No fallback or negotiate in this case.
                 await this.transport.connect(this.url, transferFormat, this);
             } else {
-                const token = this.options.accessTokenFactory();
+                const token = await this.options.accessTokenFactory();
```

Comment:
What happens if you await a string?

Reply:
Nothing good I'd imagine.

Reply:
I tested returning both promise and string and it worked just the same.

Reply:
From the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await): "If the value of the expression following the await operator is not a Promise, it's converted to a resolved Promise."

Reply:
JavaScript is weird. But hey, if it works :)